### ISSUE TITLE
Interacting with chunk loader doesn't show GUI, error in server logs. Works in singleplayer world flawlessly.

### ISSUE THREAD

**Issue by HumanMint** [TrattMods/LiterallyChunkLoader#15]
Whenever a player tries to interact with the loader, the expected GUI doesn't show up and the server console shows this message. The server doesn't crash but the same message appears when attempting to interact with the loader.

Mod version: 2.0.3 1.19.2

```[07:09:43] [Server thread/ERROR]: Error executing task on Server
java.lang.NoClassDefFoundError: net/minecraft/class_1074
	at net.literally.chunk.loader.entity.ChunkLoaderBlockEntity.method_5476(ChunkLoaderBlockEntity.java:28) ~[literally-chunk-loader-2.0.3-1.19.jar:?]
	at net.minecraft.class_3222.method_17355(class_3222.java:1047) ~[server-intermediary.jar:?]
	at net.literally.chunk.loader.block.ChunkLoaderBlock.method_9534(ChunkLoaderBlock.java:61) ~[literally-chunk-loader-2.0.3-1.19.jar:?]
	at net.minecraft.class_4970$class_4971.method_26174(class_4970.java:964) ~[server-intermediary.jar:?]
	at net.minecraft.class_3225.method_14262(class_3225.java:338) ~[server-intermediary.jar:?]
	at net.minecraft.class_3244.method_12046(class_3244.java:1145) ~[server-intermediary.jar:?]
	at net.minecraft.class_2885.method_12547(class_2885.java:34) ~[server-intermediary.jar:?]
	at net.minecraft.class_2885.method_11054(class_2885.java:8) ~[server-intermediary.jar:?]
	at net.minecraft.class_2600.method_11072(class_2600.java:22) ~[server-intermediary.jar:?]
	at net.minecraft.class_3738.run(class_3738.java:18) ~[server-intermediary.jar:?]
	at net.minecraft.class_1255.redirect$zli000$redirectExecuteTask(class_1255.java:521) ~[server-intermediary.jar:?]
	at net.minecraft.class_1255.method_16075(class_1255.java:131) ~[server-intermediary.jar:?]
	at net.minecraft.server.MinecraftServer.method_20415(MinecraftServer.java:10777) ~[server-intermediary.jar:?]
	at net.minecraft.server.MinecraftServer.method_16075(MinecraftServer.java:752) ~[server-intermediary.jar:?]
	at net.minecraft.class_1255.method_5383(class_1255.java:116) ~[server-intermediary.jar:?]
	at net.minecraft.server.MinecraftServer.method_16208(MinecraftServer.java:736) ~[server-intermediary.jar:?]
	at net.minecraft.server.MinecraftServer.method_29741(MinecraftServer.java:669) ~[server-intermediary.jar:?]
	at net.minecraft.server.MinecraftServer.method_29739(MinecraftServer.java:257) ~[server-intermediary.jar:?]
	at java.lang.Thread.run(Thread.java:833) [?:?]
Caused by: java.lang.ClassNotFoundException: net.minecraft.class_1074
	... 19 more
```
Here is another variation of the error
	
```[07:27:16] [Server thread/ERROR]: Error executing task on Server
java.lang.NoClassDefFoundError: net/minecraft/class_1074
	at net.literally.chunk.loader.entity.ChunkLoaderBlockEntity.method_5476(ChunkLoaderBlockEntity.java:28) ~[literally-chunk-loader-2.0.3-1.19.jar:?]
	at net.minecraft.class_3222.method_17355(class_3222.java:1047) ~[server-intermediary.jar:?]
	at net.literally.chunk.loader.block.ChunkLoaderBlock.method_9534(ChunkLoaderBlock.java:61) ~[literally-chunk-loader-2.0.3-1.19.jar:?]
	at net.minecraft.class_4970$class_4971.method_26174(class_4970.java:964) ~[server-intermediary.jar:?]
	at net.minecraft.class_3225.method_14262(class_3225.java:338) ~[server-intermediary.jar:?]
	at net.minecraft.class_3244.method_12046(class_3244.java:1145) ~[server-intermediary.jar:?]
	at net.minecraft.class_2885.method_12547(class_2885.java:34) ~[server-intermediary.jar:?]
	at net.minecraft.class_2885.method_11054(class_2885.java:8) ~[server-intermediary.jar:?]
	at net.minecraft.class_2600.method_11072(class_2600.java:22) ~[server-intermediary.jar:?]
	at net.minecraft.class_3738.run(class_3738.java:18) ~[server-intermediary.jar:?]
	at net.minecraft.class_1255.redirect$zli000$redirectExecuteTask(class_1255.java:521) ~[server-intermediary.jar:?]
	at net.minecraft.class_1255.method_16075(class_1255.java:131) ~[server-intermediary.jar:?]
	at net.minecraft.server.MinecraftServer.method_20415(MinecraftServer.java:10777) ~[server-intermediary.jar:?]
	at net.minecraft.server.MinecraftServer.method_16075(MinecraftServer.java:752) ~[server-intermediary.jar:?]
	at net.minecraft.class_1255.method_18857(class_1255.java:140) ~[server-intermediary.jar:?]
	at net.minecraft.server.MinecraftServer.method_16208(MinecraftServer.java:737) ~[server-intermediary.jar:?]
	at net.minecraft.server.MinecraftServer.method_29741(MinecraftServer.java:669) ~[server-intermediary.jar:?]
	at net.minecraft.server.MinecraftServer.method_29739(MinecraftServer.java:257) ~[server-intermediary.jar:?]
	at java.lang.Thread.run(Thread.java:833) [?:?]
Caused by: java.lang.ClassNotFoundException: net.minecraft.class_1074
	at jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:641) ~[?:?]
	at jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:188) ~[?:?]
	at java.lang.ClassLoader.loadClass(ClassLoader.java:520) ~[?:?]
	at org.quiltmc.loader.impl.launch.knot.KnotClassLoader.loadClass(KnotClassLoader.java:159) ~[quilt-loader-0.17.6.jar:?]
	at java.lang.ClassLoader.loadClass(ClassLoader.java:520) ~[?:?]
	... 19 more
```
The mod seems to work as expected in a singleplayer world but doesn't work on a server.

I added this mod to the server to replace another chunkloader, simple chunkloader.

A new world generated with this mod installed seems to have the same issue.

To note I am using the Quilt loader (0.17.6) in combination with Quilted Fabric API instead of fabric.

A server with just the literally chunkloader mod installed (and quilted fabric api) has the same issue but it crashes the server with these logs

```[07:37:49] [Server thread/ERROR]: Encountered an unexpected exception
java.lang.NoClassDefFoundError: net/minecraft/class_1074
	at net.literally.chunk.loader.entity.ChunkLoaderBlockEntity.method_5476(ChunkLoaderBlockEntity.java:28) ~[literally-chunk-loader-2.0.3-1.19.jar:?]
	at net.minecraft.class_3222.method_17355(class_3222.java:1047) ~[server-intermediary.jar:?]
	at net.literally.chunk.loader.block.ChunkLoaderBlock.method_9534(ChunkLoaderBlock.java:61) ~[literally-chunk-loader-2.0.3-1.19.jar:?]
	at net.minecraft.class_4970$class_4971.method_26174(class_4970.java:964) ~[server-intermediary.jar:?]
	at net.minecraft.class_3225.method_14262(class_3225.java:338) ~[server-intermediary.jar:?]
	at net.minecraft.class_3244.method_12046(class_3244.java:1145) ~[server-intermediary.jar:?]
	at net.minecraft.class_2885.method_12547(class_2885.java:34) ~[server-intermediary.jar:?]
	at net.minecraft.class_2885.method_11054(class_2885.java:8) ~[server-intermediary.jar:?]
	at net.minecraft.class_2600.method_11072(class_2600.java:22) ~[server-intermediary.jar:?]
	at net.minecraft.class_3738.run(class_3738.java:18) ~[server-intermediary.jar:?]
	at net.minecraft.class_1255.method_18859(class_1255.java:157) ~[server-intermediary.jar:?]
	at net.minecraft.class_4093.method_18859(class_4093.java:23) ~[server-intermediary.jar:?]
	at net.minecraft.server.MinecraftServer.method_24306(MinecraftServer.java:776) ~[server-intermediary.jar:?]
	at net.minecraft.server.MinecraftServer.method_18859(MinecraftServer.java:159) ~[server-intermediary.jar:?]
	at net.minecraft.class_1255.method_16075(class_1255.java:131) ~[server-intermediary.jar:?]
	at net.minecraft.server.MinecraftServer.method_20415(MinecraftServer.java:758) ~[server-intermediary.jar:?]
	at net.minecraft.server.MinecraftServer.method_16075(MinecraftServer.java:752) ~[server-intermediary.jar:?]
	at net.minecraft.class_1255.method_18857(class_1255.java:140) ~[server-intermediary.jar:?]
	at net.minecraft.server.MinecraftServer.method_16208(MinecraftServer.java:737) ~[server-intermediary.jar:?]
	at net.minecraft.server.MinecraftServer.method_29741(MinecraftServer.java:669) ~[server-intermediary.jar:?]
	at net.minecraft.server.MinecraftServer.method_29739(MinecraftServer.java:257) ~[server-intermediary.jar:?]
	at java.lang.Thread.run(Thread.java:833) [?:?]
Caused by: java.lang.ClassNotFoundException: net.minecraft.class_1074
	at jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:641) ~[?:?]
	at jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:188) ~[?:?]
	at java.lang.ClassLoader.loadClass(ClassLoader.java:520) ~[?:?]
	at org.quiltmc.loader.impl.launch.knot.KnotClassLoader.loadClass(KnotClassLoader.java:159) ~[quilt-loader-0.17.6.jar:?]
	at java.lang.ClassLoader.loadClass(ClassLoader.java:520) ~[?:?]
	... 22 more
[07:37:50] [Server thread/ERROR]: This crash report has been saved to: <filedirectory>
[07:37:50] [Server thread/INFO]: Stopping server
```

Strangely a singleplayer world with just the literally chunkloader mod (and quilted fabric api) works as expected.

Let me know if you need more details!

**Comment by HumanMint** [comment]
Just tested with a fabric server and it seems like the issue is there as well, so I guess it isn't just a quilt issue.

#14

**Comment by tratteo** [comment]
Guess a problem with some conflicts with some other mods. In #14 there are evidence of correctly working with 1.19.
Some mods may be conflicting. Other times problems were due to this reason

**Comment by HumanMint** [comment]
Thanks for the fast response!

I mentioned that I tested this mod with no other mods installed using Quilt but I guess I wasn't clear that I also tried just this mod when I mentioned I tried on Fabric as well.

To double check, I tried once again with the following setup:
New installation of fabric loader 0.14.11
mods installed:

- fabric-api-0.69.0+1.19.2.jar
- literally-chunk-loader-2.0.3-1.19.jar

Once again, the mod seems to work perfectly fine in a singleplayer world. The UI opens up as expected.
However, when I join a server with the same fabric loader and the same mods (nothing else) the server crashes with the following logs.

[Server Log](https://mclo.gs/sNu3u09)
[Crash report](https://pastebin.com/HgpZSDZa)

Since there was indeed a user that claimed it worked perfectly fine with their setup in #14, I tried to replicate their setup with the same version of Fabric Loader 0.14.9-1.19.2

Once again, the mod works as expected in a singleplayer world but crashes in a server environment when interacting with the chunk loader.

[Server Log](https://mclo.gs/wKmrhpV)
[Crash report](https://pastebin.com/wk2JPxGX)

I am a ilttle confused with the users claims since I tried to replicate their exact setup and it resulted with the same issues. I am more confused since it works perfectly fine in a singleplayer world but not a server, since from what I understand singleplayer worlds use an internal server anyways.

Let me know if you needed more information!

Thanks for your work

**Comment by tratteo** [comment]
Yes the single player world should be handled the same way as multiplayer, just with client and server packed together as host.
Hence it is very weird that it happens in a multiplayer world only.
I will try to double check methods that are called exclusively on the server, even though Minecraft obfuscation will make the job hard.
I am not very familiar with Quilt, you tested also on a vanilla Fabric environment right?

**Comment by HumanMint** [comment]
Yes the original post was made for my quilt environment but the next post I tried in a vanilla fabric environment. I don't expect you to try to fix it for quilt since I understand it is a platform that you didn't intend support for this mod.

Thank you for your fast help!

**Comment by tratteo** [comment]
The error *seems* to be related to a translation text (name of the block). Updates when i discover more

**Comment by tratteo** [comment]
Fixed in **2.0.4**
https://www.curseforge.com/minecraft/mc-mods/literally-chunk-loader/files/4193044

**Comment by lilmnm-kamikaze-** [comment]
> Fixed in **2.0.4** https://www.curseforge.com/minecraft/mc-mods/literally-chunk-loader/files/4193044

Link gives 404 error.

**Comment by tratteo** [comment]
> > Fixed in **2.0.4** https://www.curseforge.com/minecraft/mc-mods/literally-chunk-loader/files/4193044
> 
> Link gives 404 error.

Does not. Was under review

**Comment by lilmnm-kamikaze-** [comment]
> > > Fixed in **2.0.4** https://www.curseforge.com/minecraft/mc-mods/literally-chunk-loader/files/4193044
> > 
> > 
> > Link gives 404 error.
> 
> Does not. Was under review

At time of comment it was.
